bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccHandler.java

```diff
@@ -0,0 +1,547 @@
+                    refresh();
+                    break;
+                case CHANNEL_LOADPOINT_MIN_CURRENT:
+                    setMinCurrent(config.url, loadpoint, Integer.parseInt(command.toString().replaceAll(" A", "")));
+                    refresh();
+                    break;
+                case CHANNEL_LOADPOINT_MAX_CURRENT:
+                    setMaxCurrent(config.url, loadpoint, Integer.parseInt(command.toString().replaceAll(" A", "")));
+                    refresh();
+                    break;
+            }
+        }
+
+        // Note: if communication with thing fails for some reason,
+        // indicate that by setting the status with detail information:
+        // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+        // "Could not control device at IP address x.x.x.x");
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(EvccConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        if ("".equals(config.url)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, "No host configured");
+        } else {
+            // Background initialization:
+            scheduler.execute(() -> {
+                status = getStatus(config.url);
+                try {
+                    sitename = status.getResult().getSiteTitle();
+                    numberOfLoadpoints = status.getResult().getLoadpoints().length;
+                    logger.debug("Found {} loadpoints on site {} (host: {}).", numberOfLoadpoints, sitename,
+                            config.url);
+                    updateStatus(ThingStatus.ONLINE);
+                    refreshOnStartup();
+                } catch (Exception e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                            "Failed to connect to evcc: " + e);
+                }
+            });
+        }
+    }
+
+    private void statePolling() {
+        logger.debug("Running polling job ...");
+        refresh();
+    }
+
+    private void refresh() {
+        status = getStatus(config.url);
+        try {
+            sitename = status.getResult().getSiteTitle();
+            numberOfLoadpoints = status.getResult().getLoadpoints().length;
+            logger.debug("Found {} loadpoints on site {} (host: {}).", numberOfLoadpoints, sitename, config.url);
+            updateStatus(ThingStatus.ONLINE);
+            batteryConfigured = status.getResult().getBatteryConfigured();
+            gridConfigured = status.getResult().getGridConfigured();
+            pvConfigured = status.getResult().getPvConfigured();
+            updateChannelsGeneral();
+            for (int i = 0; i < numberOfLoadpoints; i++) {
+                updateChannelsLoadpoint(i);
+            }
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
```

Comment:
This seems more like `COMMUNICATION_ERROR` if related to connection failure as the cause text indicates. However, catching raw `Exception` could indicate that you may not know what actually went wrong. So I would suggest also to catch only needed, specific exception types here.

Reply:
Fixed by catching generic communication error with `null` checking, I have preserved this for any other non-communication error (I could not really imagine any for now).